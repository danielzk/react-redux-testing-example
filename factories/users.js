import {Factory} from 'rosie'
import {OrderedMap} from 'immutable'

export const UserFactory = new Factory()
  .sequence('id', sequence => sequence)
  .sequence('username', sequence => 'username' + sequence)
  .attr('firstName', 'Mark')
  .attr('lastName', 'Zucaritas')

export const UserListFactory = (count) => {
  const users = UserFactory.buildList(count)
  return OrderedMap(users.map(item => [item.id, item]))
}
