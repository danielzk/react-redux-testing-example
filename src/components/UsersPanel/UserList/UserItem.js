import React from 'react'

export default class UserItem extends React.Component {
  onDeleteClick = () => {
    this.props.actions.deleteUser(this.props.user.id)
  }

  render() {
    return (
      <li>
        <span>Username: {this.props.user.username}</span>
        <span>Full name: {this.props.user.firstName} {this.props.user.lastName}</span>
        <button onClick={this.onDeleteClick}>Delete</button>
      </li>
    )
  }
}
