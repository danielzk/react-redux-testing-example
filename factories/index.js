import {Factory} from 'rosie'

import {UserFactory, UserListFactory} from './users'

export const StoreFactory = new Factory()
  .option('numUsers', 2)
  .attr('users', ['numUsers'], count => UserListFactory(count))

export {UserFactory, UserListFactory}
