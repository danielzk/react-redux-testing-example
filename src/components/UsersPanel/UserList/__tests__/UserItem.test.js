import React from 'react'
import {shallow} from 'enzyme'

import {UserFactory} from '../../../../../factories'
import UserItem from '../UserItem'

describe('UserItem component', () => {
  it('should render username, full name and delete button', () => {
    const user = UserFactory.build()
    const wrapper = shallow(<UserItem user={user}/>)
    expect(wrapper).toMatchSnapshot()
  })

  it('should call deleteUser when delete button is clicked', () => {
    const props = {
      user:  UserFactory.build(),
      actions: {
        deleteUser: jest.fn(),
      }
    }
    const wrapper = shallow(<UserItem {...props}/>)
    wrapper.find('button').simulate('click')
    expect(props.actions.deleteUser).toBeCalledWith(props.user.id)
    expect(props.actions.deleteUser).toHaveBeenCalledTimes(1)
  })
})
