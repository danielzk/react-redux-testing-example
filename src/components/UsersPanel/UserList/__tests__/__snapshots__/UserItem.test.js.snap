// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserItem component should render username, full name and delete button 1`] = `
<li>
  <span>
    Username: 
    username1
  </span>
  <span>
    Full name: 
    Mark
     
    Zucaritas
  </span>
  <button
    onClick={[Function]}
  >
    Delete
  </button>
</li>
`;
